src/analysis/specific_limits.lean

```diff
@@ -545,6 +545,65 @@ end
 
 end summable_le_geometric
 
+section normed_ring_geometric
+variables [normed_ring α]
+
+open normed_space
+
+/- In a normed ring, the powers of an element x with `∥x∥ < 1` tend to zero.
+Proved above (same name, different namespace) for normed fields. -/
+lemma normed_ring.tendsto_pow_at_top_nhds_0_of_norm_lt_1 (x : α) (h : ∥x∥ < 1) :
+  tendsto (λ (n : ℕ), x ^ n) at_top (𝓝 0) :=
+begin
+  apply squeeze_zero_norm' (norm_pow_le_eventually x),
+  exact tendsto_pow_at_top_nhds_0_of_lt_1 (norm_nonneg _) h,
+end
+
+/- A geometric series in a complete normed ring is summable.
+Proved above (same name, different namespace) for normed fields. -/
+lemma normed_ring.summable_geometric_of_norm_lt_1
+  [complete_space α] (x : α) (h : ∥x∥ < 1) : summable (λ (n:ℕ), x ^ n) :=
+begin
+  have h1 : summable (λ (n:ℕ), ∥x∥ ^ n) := summable_geometric_of_lt_1 (norm_nonneg _) h,
+  refine summable_of_norm_bounded_eventually _ h1 _,
+  rw nat.cofinite_eq_at_top,
+  exact norm_pow_le_eventually x,
+end
+
+lemma geom_series_mul_neg [complete_space α] (x : α) (h : ∥x∥ < 1) :
+  (tsum (λ (i:ℕ), x ^ i)) * (1 - x) = 1 :=
```

Comment:
Please use notation for `tsum`. If you move these two lemmas above, then `has_sum_geometric_of_lt_1` and `tsum_geometric_of_lt_1` can be corollaries of this lemma.

Reply:
Thanks, I've made the change in notation (here and throughout).  I'm not sure whether your second recommendation is possible; see my previous comment.